### PR TITLE
upgrade poison dependency

### PR DESCRIPTION
**Pull request by andrzejsliwa**
I would like to upgrade poison dependency, I also made cleanup in tests.

next step will be updating plug for it.

### CODE REVIEW

File: test/bugsnag_test.exs [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
   use ExUnit.Case
 
+  @tag capture_log: true
   test "it doesn't raise errors if you report garbage" do
     Bugsnag.report(Enum, %{ignore: :this_error_in_test})
   end
 
   test "it handles real errors" do
     try do
-      :foo = :bar
```
Comment (andrzejsliwa) [inline test/bugsnag_test.exs]: this was issuing warning in compilation, and can be replaced by explicit raising exception

File: test/bugsnag_test.exs [hunk 2]
```diff
@@ -1,13 +1,14 @@
 defmodule BugsnagTest do
   use ExUnit.Case
 
+  @tag capture_log: true
```
Comment (andrzejsliwa) [inline test/bugsnag_test.exs]: error output goes on screen in tests, in case when was expected, so capture_log is best for it to avoid confusion.

**Comment by mootpointer** [comment]
Is there any need to remove support from Poison 1.x? Would a version requirement of `~> 1.4 or ~> 2.0` be more appropriate for best compatibility (since we don't use anything specific to the new version of Poison.

**Comment by andrzejsliwa** [comment]
The problem is that some libraries I'm using already migrated to recent version and I have conflicts. Time for forward, I verified that code is working well on it. I would keep deps updated to recent versions.

**Comment by mootpointer** [comment]
The `~> 1.4 or ~> 2.0` requirement would handle your issue. I'm in a similar position, so the above allows for 2.0 where it is possible, but doesn't cut off those who still have hard requirements for Poison 1.x off.

**Comment by mootpointer** [comment]
I've updated the poison dependency in f1fd200. It's a more liberal requirement which is still backwards compatible.

**Pull request closed without merging**